drop unsupported_reason_add

kbrock opened the pull request:
part of:
- https://github.com/ManageIQ/manageiq/pull/22898
 
Deprecating `unsupported_reason_add`. Returning the string instead

Note, these are all the same:

```ruby
unsupported_reason_add(:feature, unsupported_reason(:feature2)) unless supports_feature2?
unsupported_reason_add(:feature, unsupported_reason(:feature2)) unless supports?(:feature2)
unsupported_reason(:feature2) unless supports?(:feature2)
unsupported_reason(:feature2)
```

agrare commented on app/models/manageiq/providers/redhat/infra_manager/vm.rb:
```diff
@@ -6,9 +6,9 @@ class ManageIQ::Providers::Redhat::InfraManager::Vm < ManageIQ::Providers::Ovirt
-      unsupported_reason_add(:migrate, "Migrate operation in not supported.")
+      "Migrate operation in not supported."
     elsif !ext_management_system.supports?(:migrate)
-      unsupported_reason_add(:migrate, 'RHV API version does not support migrate')
+      'RHV API version does not support migrate'
```
Could this be?
```
else
  ext_management_system.unsupported_reason(:migrate)
```

to prevent copying the unsupported reason string in two places?

agrare commented on app/models/manageiq/providers/redhat/infra_manager/host.rb:
```diff
@@ -7,7 +7,7 @@ def provider_object(connection = nil)
 
   supports :quick_stats do
     unless ext_management_system.supports?(:quick_stats)
-      unsupported_reason_add(:quick_stats, 'RHV API version does not support quick_stats')
+      'RHV API version does not support quick_stats'
```
Same here could this be `supports(:quick_stats) { ext_management_system.unsupported_reason(:quick_stats) }`

kbrock commented on app/models/manageiq/providers/redhat/infra_manager/vm.rb:
```diff
@@ -6,40 +6,39 @@ class ManageIQ::Providers::Redhat::InfraManager::Vm < ManageIQ::Providers::Ovirt
     if blank? || orphaned? || archived?
-      unsupported_reason_add(:migrate, "Migrate operation in not supported.")
-    elsif !ext_management_system.supports?(:migrate)
-      unsupported_reason_add(:migrate, 'RHV API version does not support migrate')
+      "Migrate operation in not supported."
+    else
+      ext_management_system.unsupported_reason(:migrate)
     end
```
wasn't sure if we could get away with:

```ruby
supports :migrate do
  unsupported_reason_add(:action) || ext_management_system.unsupported_reason(:migrate)
```

Also thinking the `blank?` wants to be `ext_management_system.blank?` (basically will follow the `if ems ; (:active) || (:migrate) ; else _("no ems")` pattern

kbrock commented on app/models/manageiq/providers/redhat/infra_manager/vm.rb:
```diff
@@ -6,40 +6,39 @@ class ManageIQ::Providers::Redhat::InfraManager::Vm < ManageIQ::Providers::Ovirt
 
   supports_not :reset
   supports :publish do
     if blank? || orphaned? || archived?
```
yea. not sure what `blank?` means. think it wanted to be `ems.blank?`, which happens to be handled in this case.

agrare replied:
Yeah not sure what blank? is here either...

kbrock commented:
update:
- added 2 more deletes for unsupported_reason_add
- fixed `unless ems.supports?(feature) ; _("reason") ; end` and replaced w/ standard `ems.unsupported_reason(feature)` instead
